go/pkg/sdk/sdk_test.go

```diff
@@ -259,3 +404,126 @@ func TestSdkGetSupportedStablecoins(t *testing.T) {
+			}
+		} else {
+			assetContract, err := optimismContracts.NewSavingsDaiCaller(asset, client)
+			if err != nil {
+				return nil, fmt.Errorf("failed to load SavingsDai contract: %w", err)
+			}
+			balanceAsset, err = assetContract.BalanceOf(nil, from)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get sDai user's balance: %v", err)
+			}
+		}
+	} else if chainId == 10 {
+		if isDeposit {
+			assetContract, err := optimismContracts.NewFiatTokenV22Caller(asset, client)
+			if err != nil {
+				return nil, fmt.Errorf("failed to load FiatTokenV22 contract: %w", err)
+			}
+			balanceAsset, err = assetContract.BalanceOf(nil, from)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get USDC user's balance: %v", err)
+			}
+
+		} else {
+			assetContract, err := optimismContracts.NewSavingsDaiCaller(asset, client)
+			if err != nil {
+				return nil, fmt.Errorf("failed to load SavingsDai contract: %w", err)
+			}
+			balanceAsset, err = assetContract.BalanceOf(nil, from)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get sDai user's balance: %v", err)
+			}
+		}
+	}
+	return balanceAsset, nil
+}
+
+func increaseAllowance(from common.Address, spender common.Address, token common.Address, amount *big.Int, isDeposit bool, privateKey *ecdsa.PrivateKey, rpcEndpoint string, chainId int64) (*types.Transaction, error) {
+	var asset abi.ABI
+	var err error
+
+	if chainId == 1 {
+		if isDeposit {
+			asset, err = abi.JSON(strings.NewReader(ethereumContracts.DaiABI))
+			if err != nil {
+				log.Fatalf("Failed to parse Dai ABI: %v", err)
+			}
+		} else {
+			asset, err = abi.JSON(strings.NewReader(ethereumContracts.SavingsDaiABI))
+			if err != nil {
+				log.Fatalf("Failed to parse sDai ABI: %v", err)
+			}
+		}
+	} else if chainId == 10 {
+		if isDeposit {
+			asset, err = abi.JSON(strings.NewReader(optimismContracts.FiatTokenV22ABI))
+			if err != nil {
+				log.Fatalf("Failed to parse FiatTokenV22 ABI: %v", err)
+			}
+		} else {
+			asset, err = abi.JSON(strings.NewReader(optimismContracts.SavingsDaiABI))
+			if err != nil {
+				log.Fatalf("Failed to parse sDai ABI: %v", err)
+			}
+		}
+	}
+
+	data, err := asset.Pack("approve", spender, amount)
+	if err != nil {
+		log.Fatalf("Failed to pack data for transaction: %v", err)
+	}
+
+	tx, err := utils.CreateTransaction(from, token, data, rpcEndpoint)
+	if err != nil {
+		log.Fatalf("Increase allowance failed: %v", err)
+	}
+	fmt.Println("- Transaction created")
+
+	signedTx, err := utils.SignTransaction(tx, big.NewInt(chainId), privateKey)
+	if err != nil {
+		log.Fatalf("Increase allowance failed: %v", err)
+	}
+	fmt.Println("-- Transaction signed")
+
+	err = utils.SendTransaction(signedTx, rpcEndpoint)
+	if err != nil {
+		log.Fatalf("Increase allowance failed: %v", err)
+	}
+	fmt.Println("---- Transaction sent")
+
+	return signedTx, nil
+}
+
+func scaleTo6decimals(roundedExpectedReceivedSDai *big.Int, scaleFactor *big.Float) *big.Int {
```

Comment:
misleading name since the number 6 doesn't appear in the code. rename it. also you can use simpler variables names like `amount` and `scaledAmount` since this function doesn't depend on sDai but is generic